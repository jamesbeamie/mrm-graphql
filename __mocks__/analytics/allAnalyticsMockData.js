export default {
  analytics: {
    bookingsCount: [
      {
        totalBookings: 0,
        period: 'Jan 2018',
      },
      {
        totalBookings: 0,
        period: 'Feb 2018',
      },
      {
        totalBookings: 0,
        period: 'Mar 2018',
      },
      {
        totalBookings: 0,
        period: 'Apr 2018',
      },
      {
        totalBookings: 0,
        period: 'May 2018',
      },
      {
        totalBookings: 0,
        period: 'Jun 2018',
      },
      {
        totalBookings: 0,
        period: 'Jul 2018',
      },
      {
        totalBookings: 0,
        period: 'Aug 2018',
      },
      {
        totalBookings: 0,
        period: 'Sep 2018',
      },
      {
        totalBookings: 0,
        period: 'Oct 2018',
      },
      {
        totalBookings: 0,
        period: 'Nov 2018',
      },
      {
        totalBookings: 0,
        period: 'Dec 2018',
      },
      {
        totalBookings: 0,
        period: 'Jan 2019',
      },
      {
        totalBookings: 0,
        period: 'Feb 2019',
      },
      {
        totalBookings: 0,
        period: 'Mar 2019',
      },
      {
        totalBookings: 0,
        period: 'Apr 2019',
      },
      {
        totalBookings: 0,
        period: 'May 2019',
      },
    ],
    checkinsPercentage: 0.0,
    appBookingsPercentage: 0.0,
    cancellationsPercentage: 0.0,
    autoCancellationsPercentage: 0.0,
    analytics: [
      {
        roomName: 'Kampala',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Kampala2',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'New York',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Cafeteria',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'China Town',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Hogwarts',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Kigali',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Krypton',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Pich',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Rabat',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Rich Port',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Stone Henge',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Room1',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Tent',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Tokyo',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Younde',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Valhalla',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
    ],
    rooms: [
      {
        id: '6',
        name: 'Cafeteria',
        capacity: 64,
      },
      {
        id: '3',
        name: 'China Town',
        capacity: 4,
      },
      {
        id: '5',
        name: 'Hogwarts',
        capacity: 64,
      },
      {
        id: '18',
        name: 'Kampala',
        capacity: 14,
      },
      {
        id: '4',
        name: 'Kampala2',
        capacity: 4,
      },
      {
        id: '14',
        name: 'Kigali',
        capacity: 30,
      },
      {
        id: '13',
        name: 'Krypton',
        capacity: 30,
      },
      {
        id: '2',
        name: 'New York',
        capacity: 4,
      },
      {
        id: '11',
        name: 'Pich',
        capacity: 50,
      },
      {
        id: '8',
        name: 'Rabat',
        capacity: 14,
      },
      {
        id: '12',
        name: 'Rich Port',
        capacity: 10,
      },
      {
        id: '17',
        name: 'Room1',
        capacity: 6,
      },
      {
        id: '15',
        name: 'Stone Henge',
        capacity: 20,
      },
      {
        id: '7',
        name: 'Tent',
        capacity: 164,
      },
      {
        id: '16',
        name: 'Tokyo',
        capacity: 20,
      },
      {
        id: '9',
        name: 'Valhalla',
        capacity: 14,
      },
      {
        id: '10',
        name: 'Younde',
        capacity: 14,
      },
    ],
  },
  fetching: false,
};
