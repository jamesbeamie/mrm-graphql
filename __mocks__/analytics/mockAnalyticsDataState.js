const mockDataForAnalytics = {
  checkinsAndCancellations: {
    bookings: 42,
    cancellations: 0,
    cancellationsPercentage: 0,
    checkins: 0,
    checkinsPercentage: 0,
  },
  averageMeetingDuration: [
    0, 0, 0, 0,
  ],
  averageMeetingTime: {
    MeetingsDurationaAnalytics: [
      {
        roomName: 'Test 1',
        count: 4,
        totalDuration: 25,
      },
      {
        roomName: 'Test 2',
        count: 1,
        totalDuration: 50,
      },
      {
        roomName: 'Test 3',
        count: 1,
        totalDuration: 35,
      },
      {
        roomName: 'Test 4',
        count: 1,
        totalDuration: 2440,
      },
    ],
  },
  mostBookedRooms: [
    {
      meetings: 7,
      percentage: 100,
      roomName: 'Moroto',
    },
  ],
  leastBookedRooms: [
    {
      meetings: 1,
      percentage: 100,
      roomName: 'Moroto B',
    },
  ],
  roomCapacity:
    {
      lessThanTenData: 96,
      betweenTenandTwentyData: 4,
      greaterThanTwentyData: 0,
    },
  totalBookingsCount: [
    { period: 'Mar 18 2019', bookings: 17 },
    { period: 'Mar 17 2019', bookings: 14 },
  ],
};
export default mockDataForAnalytics;
