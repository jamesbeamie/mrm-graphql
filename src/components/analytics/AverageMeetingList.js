import React, { useContext, useState } from 'react';
import '../../assets/styles/averageMeetinglist.scss';
import TableHead from '../helpers/TableHead';
import QueryAnalyticsPerMeetingRoom from './AverageMeetingList/QueryAnalyticsPerMeetingRoom';
import Tip from '../commons/Tooltip';
import Pagination from '../commons/Pagination';
import Overlay from '../commons/Overlay';
import ErrorIcon from '../commons/ErrorIcon';
import meetingDurationAnalyticsMock from '../../fixtures/meetingDurationAnalytics';
import AnalyticsContext from '../helpers/AnalyticsContext';
import getMeetingDurationAnalytics from '../helpers/analytics/AverageMeetingList';
import paginate from '../helpers/FrontendPagination';

export const AverageMeetingList = () => {
  const { fetching, analytics } = useContext(AnalyticsContext);
  const [{ perPage, currentPage }, updatePageDetails] = useState({ perPage: 5, currentPage: 1 });

  const tip =
     'The number of meetings in a room,  the average number of attendees to these meetings as well as the average duration of the meetings.';

  const analyticsForMeetingsDurationsMock = {
    meetingDurationAnalytics: meetingDurationAnalyticsMock(3),
  };

  const error = analytics === null;

  let analyticsForMeetingsDurations;

  if (!fetching) {
    analyticsForMeetingsDurations = paginate(
      analytics.analytics,
      { currentPage }, {
        formatter: getMeetingDurationAnalytics,
        dataName: 'meetingDurationAnalytics',
      },
    );
  }

  const analyticsData = fetching
    ? analyticsForMeetingsDurationsMock
    : analyticsForMeetingsDurations;


  const handlePageChange = (perPageData, currentPageData) => updatePageDetails({
    perPage: perPageData,
    currentPage: currentPageData,
  });

  const displayPaginator = !fetching
  && !(analyticsData.meetingDurationAnalytics.length < 5 && currentPage === 1);

  return (
    <div className="average-meeting overlay-container">
      {fetching && <Overlay />}
      <div className="average-meeting-control">
        <h4 className="header-title">Average time spent/Meeting Room</h4>
        <span className="moreVerticalIcon">{Tip(tip)}</span>
      </div>
      <div className="average-meeting-list">
        <div>
          <TableHead
            titles={['Room', 'No. of meetings', 'Average Meeting Duration']}
          />
          <div>
            {
              error ? (
                <ErrorIcon
                  message="No resource found"
                />
              )
              : <QueryAnalyticsPerMeetingRoom
                data={analyticsData}
              />
            }
          </div>
        </div>
      </div>
      {displayPaginator &&
        <div className="average-meeting-pagination">
          <div>
            <Pagination
              totalPages={analyticsForMeetingsDurations.pages}
              hasNext={analyticsForMeetingsDurations.hasNext}
              hasPrevious={analyticsForMeetingsDurations.hasPrevious}
              handleData={handlePageChange}
              isFetching={fetching}
              perPage={perPage}
              currentPage={currentPage}
            />
          </div>
        </div>
      }
    </div>
  );
};

export default AverageMeetingList;
