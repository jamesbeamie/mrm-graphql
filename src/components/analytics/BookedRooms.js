/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import TableHead from '../helpers/TableHead';
import '../../assets/styles/bookedroom.scss';
import { pollRed } from '../../utils/images/images';
import Tip from '../commons/Tooltip';
import ErrorIcon from '../commons/ErrorIcon';
import Overlay from '../commons/Overlay';

/**
 *
 * Component for showing booked rooms
 *
 * @param {Object} bookedRoomsObject
 *
 * @returns {JSX}
 */
const BookedRooms = ({
  bookedRoomText, bookedRoomsList, fetching, tip,
}) => (
  <div className="wrap-booked-room overlay-container">
    {fetching && <Overlay />}
    <div className="booked-room-header">
      <img src={pollRed} alt="Pull" />
      <h4>{bookedRoomText}</h4>
      {Tip(tip)}
    </div>
    <div className="booked-room-list">
      <div className="table">
        <TableHead titles={['Room', 'Meetings', '% Share of All Meetings']} />
        {bookedRoomsList.length === 0 && !fetching ? (
          <div className="table__body">
            {<ErrorIcon message={bookedRoomsList.length === 0 && 'No resource found'} />}
          </div>
        ) : (
          <div className="table__body">
            {
              (fetching ? [{
                roomName: 'Room name',
                meetings: 'Meetings',
                percentage: '',
              }] : bookedRoomsList).map((room, index) => (
                <div className="table__row--analytics" key={index}>
                  <span>{room.roomName}</span>
                  <span>{room.numberOfBookings}</span>
                  <span>{Math.round(room.bookingsPercentageShare)}%</span>
                </div>
              ))
            }
          </div>
        )}
      </div>
    </div>
  </div>
);

BookedRooms.propTypes = {
  bookedRoomText: PropTypes.string.isRequired,
  tip: PropTypes.string.isRequired,
  bookedRoomsList: PropTypes.instanceOf(Array),
  fetching: PropTypes.bool.isRequired,
};

BookedRooms.defaultProps = {
  bookedRoomsList: [],
};
export default BookedRooms;
