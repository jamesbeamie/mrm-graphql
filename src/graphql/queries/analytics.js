import gql from 'graphql-tag';

const ALL_ANALYTICS = gql`
  query allAnalytics($startDate: String!, $endDate: String!, $locationId: Int!) {
    allAnalytics(startDate: $startDate, endDate: $endDate, locationId: $locationId) {
      bookingsCount {
        totalBookings
        period
      }
      checkinsPercentage
      appBookingsPercentage
      cancellationsPercentage
      autoCancellationsPercentage
      analytics {
        roomName
        cancellations
        cancellationsPercentage
        autoCancellations
        numberOfBookings
        checkins
        checkinsPercentage
        bookingsPercentageShare
        appBookings
        appBookingsPercentage
        events {
          durationInMinutes
        }
      }
    }
  }
`;

export const ANALYTICS_FOR_DAILY_ROOM_EVENTS = gql`
  query allEvents($startDate: String!, $endDate: String!, $page: Int, $perPage: Int) {
    allEvents(startDate: $startDate, endDate: $endDate, page: $page, perPage: $perPage) {
        events {
        room {
          name
        }
        eventTitle
        endTime
        checkedIn
        startTime
        checkInTime
        endTime
        cancelled
        numberOfParticipants
      }
      pages
      queryTotal
      hasNext
      hasPrevious
    }
  }
`;

export default ALL_ANALYTICS;
