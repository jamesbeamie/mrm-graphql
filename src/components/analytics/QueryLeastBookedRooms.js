import React, { useContext } from 'react';
import BookedRooms from './BookedRooms';
import AnalyticsContext from '../helpers/AnalyticsContext';

export const QueryLeastBookedRooms = () => {
  const { analytics, fetching } = useContext(AnalyticsContext);
  return (
    <div>
      <BookedRooms
        tip="The least number of times meeting rooms were booked in a set time period"
        bookedRoomText="Least Booked Rooms"
        fetching={fetching}
        bookedRoomsList={fetching ? undefined : analytics.analytics.slice(-4).reverse()}
      />
    </div>
  );
};

export default QueryLeastBookedRooms;
