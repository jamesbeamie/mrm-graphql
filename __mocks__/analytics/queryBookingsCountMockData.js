import { ANALYTICS_BOOKINGS_COUNT } from '../../src/graphql/queries/analytics';
import { getTodaysDate } from '../../src/utils/Utilities';

export const queryBookingsCountMockData = [
  {
    request: {
      query: ANALYTICS_BOOKINGS_COUNT,
      variables: {
        startDate: getTodaysDate(),
        endDate: getTodaysDate(),
      },
      result: {
        data: [],
        errors: null,
      },
    },
  },
];

export const queryBookingsCountMocksData = {
  request: {
    query: ANALYTICS_BOOKINGS_COUNT,
    variables: { startDate: 'Nov 01 2018', endDate: 'Nov 03 2018' },
  },
  result: {
    data: {
      bookingsAnalyticsCount: [
        {
          period: 'Nov 01 2018',
          bookings: 21,
        },
        {
          period: 'Nov 02 2018',
          bookings: 30,
        },
        {
          period: 'Nov 03 2018',
          bookings: 33,
        },
      ],
    },
    errors: null,
  },
};

export const queryAllRoomsCapacity = {
  betweenTenandTwentyData: 0,
  greaterThanTwentyData: 0,
  lessThanTenData: 100,
};
