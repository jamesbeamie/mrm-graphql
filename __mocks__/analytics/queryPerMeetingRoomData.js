import { MEETING_DURATION_ANALYTICS } from '../../src/graphql/queries/analytics';

const queryPerMeetingRoomData = [
  {
    request: {
      query: MEETING_DURATION_ANALYTICS,
      variables: { startDate: '01 Nov 2018', endDate: '02 Nov 2018' },
    },
    result: {
      data: {
        analyticsForMeetingsDurations: {
          MeetingsDurationaAnalytics: [
            {
              roomName: 'Tortuga',
              count: 1,
              totalDuration: 61,
            },
          ],
        },
      },
      errors: {
        graphQLErrors: [{ message: 'Error occured' }],
      },
    },
  },
];

export { queryPerMeetingRoomData as default };
