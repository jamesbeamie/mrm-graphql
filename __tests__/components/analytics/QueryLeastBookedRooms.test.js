import React from 'react';
import { mount } from 'enzyme';
import { QueryLeastBookedRooms } from '../../../src/components/analytics/QueryLeastBookedRooms';
import AnalyticsContext from '../../../src/components/helpers/AnalyticsContext';
import allAnalyticsMockData from '../../../__mocks__/analytics/allAnalyticsMockData';

jest.mock('../../../src/components/helpers/QueriesHelpers');

describe('Query Least Booked Rooms Component', () => {
  it('should load progressBar', () => {
    const wrapper = mount(
      <AnalyticsContext.Provider value={allAnalyticsMockData} >
        <QueryLeastBookedRooms />
      </AnalyticsContext.Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
