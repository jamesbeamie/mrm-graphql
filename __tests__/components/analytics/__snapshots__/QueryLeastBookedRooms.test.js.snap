// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query Least Booked Rooms Component should load progressBar 1`] = `
<QueryLeastBookedRooms>
  <div>
    <BookedRooms
      bookedRoomText="Least Booked Rooms"
      bookedRoomsList={
        Array [
          Object {
            "appBookings": 0,
            "appBookingsPercentage": 0,
            "autoCancellations": 0,
            "bookingsPercentageShare": 0,
            "cancellations": 0,
            "cancellationsPercentage": 0,
            "checkins": 0,
            "checkinsPercentage": 0,
            "events": Array [
              Object {
                "durationInMinutes": 0,
              },
            ],
            "numberOfBookings": 0,
            "roomName": "Valhalla",
          },
          Object {
            "appBookings": 0,
            "appBookingsPercentage": 0,
            "autoCancellations": 0,
            "bookingsPercentageShare": 0,
            "cancellations": 0,
            "cancellationsPercentage": 0,
            "checkins": 0,
            "checkinsPercentage": 0,
            "events": Array [
              Object {
                "durationInMinutes": 0,
              },
            ],
            "numberOfBookings": 0,
            "roomName": "Younde",
          },
          Object {
            "appBookings": 0,
            "appBookingsPercentage": 0,
            "autoCancellations": 0,
            "bookingsPercentageShare": 0,
            "cancellations": 0,
            "cancellationsPercentage": 0,
            "checkins": 0,
            "checkinsPercentage": 0,
            "events": Array [
              Object {
                "durationInMinutes": 0,
              },
            ],
            "numberOfBookings": 0,
            "roomName": "Tokyo",
          },
          Object {
            "appBookings": 0,
            "appBookingsPercentage": 0,
            "autoCancellations": 0,
            "bookingsPercentageShare": 0,
            "cancellations": 0,
            "cancellationsPercentage": 0,
            "checkins": 0,
            "checkinsPercentage": 0,
            "events": Array [
              Object {
                "durationInMinutes": 0,
              },
            ],
            "numberOfBookings": 0,
            "roomName": "Tent",
          },
        ]
      }
      fetching={false}
      tip="The least number of times meeting rooms were booked in a set time period"
    >
      <div
        className="wrap-booked-room overlay-container"
      >
        <div
          className="booked-room-header"
        >
          <img
            alt="Pull"
            src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fpoll_red.svg?alt=media&token=f162d826-43a7-4bcb-aac8-0077ea6a4885"
          />
          <h4>
            Least Booked Rooms
          </h4>
          <ThemedTooltippedComponent
            className=""
            composeTheme="deeply"
            mapThemrProps={[Function]}
            theme={null}
            tooltip="The least number of times meeting rooms were booked in a set time period"
            tooltipDelay={0}
            tooltipHideOnClick={true}
            tooltipPosition="top"
            tooltipShowOnClick={true}
          >
            <TooltippedComponent
              className=""
              theme={
                Object {
                  "tooltip": "tooltip",
                  "tooltipInner": "tooltipInner",
                }
              }
              tooltip="The least number of times meeting rooms were booked in a set time period"
              tooltipDelay={0}
              tooltipHideOnClick={true}
              tooltipPosition="top"
              tooltipShowOnClick={true}
            >
              <ThemedLink
                active={false}
                className=""
                composeTheme="deeply"
                mapThemrProps={[Function]}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                theme={
                  Object {
                    "tooltip": "tooltip",
                    "tooltipInner": "tooltipInner",
                  }
                }
              >
                <Link
                  active={false}
                  className=""
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  theme={
                    Object {
                      "tooltip": "tooltip",
                      "tooltipInner": "tooltipInner",
                    }
                  }
                >
                  <a
                    className=""
                    data-react-toolbox="link"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <img
                      alt="help icon"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
                    />
                  </a>
                </Link>
              </ThemedLink>
            </TooltippedComponent>
          </ThemedTooltippedComponent>
        </div>
        <div
          className="booked-room-list"
        >
          <div
            className="table"
          >
            <TableHead
              titles={
                Array [
                  "Room",
                  "Meetings",
                  "% Share of All Meetings",
                ]
              }
            >
              <div
                className="table__headers"
              >
                <span
                  key="Room"
                >
                  Room
                </span>
                <span
                  key="Meetings"
                >
                  Meetings
                </span>
                <span
                  key="% Share of All Meetings"
                >
                  % Share of All Meetings
                </span>
              </div>
            </TableHead>
            <div
              className="table__body"
            >
              <div
                className="table__row--analytics"
                key="0"
              >
                <span>
                  Valhalla
                </span>
                <span>
                  0
                </span>
                <span>
                  0
                  %
                </span>
              </div>
              <div
                className="table__row--analytics"
                key="1"
              >
                <span>
                  Younde
                </span>
                <span>
                  0
                </span>
                <span>
                  0
                  %
                </span>
              </div>
              <div
                className="table__row--analytics"
                key="2"
              >
                <span>
                  Tokyo
                </span>
                <span>
                  0
                </span>
                <span>
                  0
                  %
                </span>
              </div>
              <div
                className="table__row--analytics"
                key="3"
              >
                <span>
                  Tent
                </span>
                <span>
                  0
                </span>
                <span>
                  0
                  %
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </BookedRooms>
  </div>
</QueryLeastBookedRooms>
`;
